### PR TITLE
Updated README; added more test cases for new functionality around g…

### PR DESCRIPTION
**Pull request by rolivieri**
…ovendor.

This PR addresses this issue: https://github.com/IBM-Blockchain-Starter-Kit/build-lib/issues/87

### CODE REVIEW

File: test/go-chaincode/vendor-dependencies.bats
Comment (jt-nti) [inline test/go-chaincode/vendor-dependencies.bats]: Stubbing govendor twice doesn't seem right. I'm not sure how the build passed but this test is failing for me when I run it locally. I'd like to figure out what's going on before merging.

**Review by jt-nti — CHANGES_REQUESTED** [review]

File: test/go-chaincode/vendor-dependencies.bats
Comment (jt-nti) [reply test/go-chaincode/vendor-dependencies.bats]: So I think this should be something like...

```
    stub govendor \
      "init : true" \
      "fetch *github.com/hyperledger/fabric/core/chaincode/lib/cid@v1.2.1* : true" 
```

And it would be good to add a test when there's a couple of vendored packages as well

Comment (jt-nti) [reply test/go-chaincode/vendor-dependencies.bats]: So I think this should be something like...

```
    stub govendor \
      "init : true" \
      "fetch *github.com/hyperledger/fabric/core/chaincode/lib/cid@v1.2.1* : true" 
```

And it would be good to add a test when there's a couple of vendored packages as well.

The reason the test was failing locally for me is that I didn't have `readarray` installed. I think it would be better to use the `read` builtin instead, i.e. replace this...

```
        readarray packages < .govendor_packages
```

with this...

```
        while read package; do
            packages+=($package)
        done < .govendor_packages
```

File: src/go-chaincode/vendor-dependencies.sh
Comment (jt-nti) [inline src/go-chaincode/vendor-dependencies.sh]: Can this comment be deleted?

**Comment by jt-nti** [comment]
Could you also add this to the `test/go-chaincode/vendor-dependencies.bats` file please?

```
teardown() {
    cleanup_stubs
}
```

File: src/go-chaincode/vendor-dependencies.sh
Comment (rolivieri) [reply src/go-chaincode/vendor-dependencies.sh]: @jt-nti Sure, done.

File: test/go-chaincode/vendor-dependencies.bats
Comment (rolivieri) [reply test/go-chaincode/vendor-dependencies.bats]: @jt-nti  Regrading readarray and jq, you may have missed this in the README -> https://github.com/IBM-Blockchain-Starter-Kit/build-lib/pull/88/files#diff-04c6e90faac2675aa89e2176d2eec7d8R16

There was already a dependency on jq in the codebase. readarray is available by default as long as you are using bash 4 (or higher). Less code in our codebase is less code we have to maintain, debug, update, etc. : )

**Comment by rolivieri** [comment]
@jt-nti New travis ci build with latest changes to address points above just completed successfully.

**Review by jorgedr94 — APPROVED** [review]
LGTM. Merging.

**Review by jt-nti — APPROVED** [review]
I take your point about trying to minimise code but prereqs have their own drawbacks so it's a trade off- while jq definitely clears the threshold, I think I'd still like to avoid adding the readarray dependency. I'm blocked on something else at the moment, so happy to merge what you have so far and create a PR to switch to take out readarray afterwards.

**Pull request merged**
